src/fluree/db/api/query.cljc

```diff
@@ -75,38 +77,64 @@
      ;; return promise chan immediately
      pc)))
 
-(defn t-flakes->json-ld
-  [db compact cache fuel error-ch t-flakes]
+
+(defn s-flakes->json-ld
+  [db cache compact fuel error-ch s-flakes]
   (async/go
     (try*
-      (let [assert-flakes  (not-empty (filter flake/op t-flakes))
-            retract-flakes (not-empty (filter (complement flake/op) t-flakes))
-
-            asserts-chan   (json-ld-resp/flakes->res db cache compact fuel 1000000
-                                                     {:wildcard? true, :depth 0}
-                                                     0 assert-flakes)
-            retracts-chan  (json-ld-resp/flakes->res db cache compact fuel 1000000
-                                                     {:wildcard? true, :depth 0}
-                                                     0 retract-flakes)
-
-            asserts (<? asserts-chan)
-            retracts (<? retracts-chan)
-
-            ;; t is always positive for users
-            result         (cond-> {:t (- (flake/t (first t-flakes)))}
-                             asserts (assoc :assert asserts)
-                             retracts (assoc :retract retracts))]
-        result)
+      (let [json-chan (json-ld-resp/flakes->res db cache compact fuel 1000000
+                                                {:wildcard? true, :depth 0}
+                                                0 s-flakes)]
+        (-> (<? json-chan)
+            ;; add the id in case the iri flake isn't present in s-flakes
+            (assoc :id (json-ld/compact (<? (dbproto/-iri db (flake/s (first s-flakes)))) compact))))
       (catch* e
               (log/error e "Error converting history flakes.")
               (async/>! error-ch e)))))
 
+(defn t-flakes->json-ld
+  [db compact cache fuel error-ch t-flakes]
+  (go-try
+    (let [assert-flakes (not-empty (filter flake/op t-flakes))
+          retract-flakes (not-empty (filter (complement flake/op) t-flakes))
+
+          s-asserts-ch (->> (sort-by flake/s assert-flakes)
```

Comment:
This might not matter all that much in practice, but it's faster to use `group-by` instead of `sort-by` + `partition-by`, unless you care that the groups are also sorted by subject in the end.

```clojure
(->> assert-flakes
     (group-by flake/s)
     vals
     async/to-chan!)
```